packages/dds/map/src/test/directoryOracle.ts

```diff
@@ -0,0 +1,147 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "node:assert";
+
+import type { IEventThisPlaceHolder } from "@fluidframework/core-interfaces";
+
+import type {
+	IDirectory,
+	IDirectoryValueChanged,
+	ISharedDirectory,
+	IValueChanged,
+} from "../interfaces.js";
+
+/**
+ * Oracle for directory
+ * @internal
+ */
+export class SharedDirectoryOracle {
+	private readonly model = new Map<string, unknown>();
+
+	public constructor(private readonly sharedDir: ISharedDirectory) {
+		this.sharedDir.on("valueChanged", this.onValueChanged);
+		this.sharedDir.on("clear", this.onClear);
+		this.sharedDir.on("subDirectoryCreated", this.onSubDirCreated);
+		this.sharedDir.on("subDirectoryDeleted", this.onSubDirDeleted);
+		this.sharedDir.on("containedValueChanged", this.onContainedValueChanged);
+
+		this.takeSnapshot(sharedDir);
+	}
+
+	private takeSnapshot(dir: ISharedDirectory | IDirectory): void {
+		for (const [k, v] of this.sharedDir.entries()) {
+			this.model.set(k, v);
+		}
+	}
+
+	private readonly onValueChanged = (
+		changed: IDirectoryValueChanged,
+		local: boolean,
+		target: IEventThisPlaceHolder,
+	): void => {
+		// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+		const { path, key, previousValue } = changed;
+		const fullPath = path === "/" ? `/${key}` : `${path}/${key}`;
+
+		if (this.model.has(fullPath)) {
+			const prevVal = this.model.get(fullPath);
+			assert.strictEqual(
+				prevVal,
+				previousValue,
+				`previous value mismatch at ${fullPath}: expected: ${prevVal}, actual: ${previousValue}`,
+			);
+		}
+
+		// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+		const newVal = this.sharedDir.get(fullPath);
```

Comment:
this seems wrong. i don't think you can call get a full path like this, i think you need to find the directory with getWorkingDirectory, and then get the specific key off that sub-directory